src/core.js

```diff
@@ -227,6 +227,8 @@ const alignWithDOM = function(nodeName, key) {
     const keyNode = keyMap[key];
     if (matches(nodeName, key, keyNode)) {
       node = keyNode;
+    } else if (keyNode && keyNode !== currentNode) {
+      removeChild(currentParent, keyNode, keyMap);
```

Comment:
Oh I see, line 260 runs whenever the current node has a key and won't be used, but it doesn't necessarily mean that it was replaced with another node with the same key, such as here. I'll update this.